spring-tx/src/main/java/org/springframework/dao/support/DataAccessUtils.java

```diff
@@ -56,6 +61,109 @@
 		return results.iterator().next();
 	}
 
+	/**
+	 * Return a single result object from the given Stream.
+	 * <p>Returns {@code null} if 0 result objects found;
+	 * throws an exception if more than 1 element found.
+	 * @param results the result Stream (can be {@code null})
+	 * @return the single result object, or {@code null} if none
+	 * @throws IncorrectResultSizeDataAccessException if more than one
+	 * element has been found in the given Stream
+	 */
+	@Nullable
+	public static <T> T singleResult(@Nullable Stream<T> results) throws IncorrectResultSizeDataAccessException {
+		if (results == null) {
+			return null;
+		}
+		List<T> resultList = results.toList();
```

Comment:
Hi @marschall,

thanks for reviewing, sounds good, i refactored this. Does it make more sense now?

`throw new IncorrectResultSizeDataAccessException(1, resultList.size());`
as we limited the resultList size with .limit, this would not report the original passed Stream results size though but instead a maximum of 2... is that an issue?

Reply:
Yes, looks better to me.